libapicore/ApiServer.cpp

```diff
@@ -17,17 +17,37 @@ void ApiServer::start()
 	if (m_portnumber == 0) return;
 
 	m_running.store(true, std::memory_order_relaxed);
-	m_acceptor.bind(tcp::endpoint(tcp::v4(), m_portnumber));
 
-	cnote << "Api server listening for connections on port " + to_string(m_acceptor.local_endpoint().port());
+	tcp::endpoint endpoint(tcp::v4(), m_portnumber);
 
+	// Try to bind to port number
+	// if exception occurs it may be due to the fact that
+	// requested port is already in use by another service
+	try
+	{
+		m_acceptor.open(endpoint.protocol());
+		m_acceptor.bind(endpoint);
+		m_acceptor.listen(64);
+	}
+	catch (const std::exception& _e)
```

Comment:
@AndreaLanfranchi: `_e` is unreferenced here; MSVC complains about it :)

Reply:
It's a very strange complaint as it's quite common standard to catch exception by reference. Do you mean it wants it by value ?

Reply:
http://www.cplusplus.com/doc/tutorial/exceptions/

Reply:
See the CI https://ci.appveyor.com/project/ethereum-mining/ethminer/build/job/9573u5xgtp57xatv#L974

Reply:
Maybe just using one of MSVC's macros like `UNREFERENCED_PARAMETER(_e);` should do the trick.

Reply:
Will the directive be accepted by other compilers ? Or should I enclose it in a conditional block if _WIN32 ?

Reply:
This macro is MSVC specific, so we probably need to wrap it.

Reply:
Just omit the name `_e`.

Reply:
@chfast approach is commonly used.

Or, you can just add a "(void)_ec;" statement in the block. It doesn't generate any machine code and will satisfy the compiler.